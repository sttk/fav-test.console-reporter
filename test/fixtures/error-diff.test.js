'use strict';

var assert = require('assert');

describe('error-diff', function() {

  it('compare same type', function() {
    assert.equal(true, false);
  });

  it('compare different type', function() {
    assert.equal(true, 'true');
  });

  it('compare same content objects', function() {
    var o1 = { a: 1 };
    var o2 = { a: 1 };
    assert.equal(o1, o2);
  });

  it('compare different content objects deeply', function() {
    var o1 = { a: 1 };
    var o2 = { a: 2 };
    assert.deepEqual(o1, o2);
  });

  it('display add diff', function() {
    var o1 = { a: 1 };
    var o2 = { a: 1, b: 2, c: 3 };
    assert.deepEqual(o1, o2);
  });

  it('display delete diff', function() {
    var o1 = { a: 1, b: 2, c: 3 };
    var o2 = { a: 1 };
    assert.deepEqual(o1, o2);
  });

  it('different multiple blocks', function() {
    var o1 = {
      a: true,
      b: {
        c: 'A',
        d: 1,
        f: {
          g: -23,
          i: 'A',
          j: [1, 2, 3],
        },
      },
      k: 1,
      l: 2,
      m: 3,
      n: [4, 5],
    };
    var o2 = {
      a: false,
      b: {
        c: 'A',
        d: 1,
        f: {
          g: -23,
          i: 'A',
          j: [1, 2, 3],
        },
      },
      k: 1,
      l: 2,
      m: 3,
      n: [4, 6],
    };
    assert.deepEqual(o1, o2);
  });

  it('different multiple blocks but nearly', function() {
    var o1 = { a: true, b: { c: 'ABC', d: 1 } };
    var o2 = { a: false, b: { c: 'aaa', d: 1 } };
    assert.deepEqual(o1, o2);
  });
});
