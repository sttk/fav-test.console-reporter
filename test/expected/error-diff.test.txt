

  error-diff
    1) compare same type
    2) compare different type
    3) compare same content objects
    4) compare different content objects deeply
    5) display add diff
    6) display delete diff
    7) different multiple blocks
    8) different multiple blocks but nearly


  0 passing (14ms)
  8 failing

  1) error-diff
       compare same type:

      AssertionError: ==
      + expected - actual

      -true
      +false

      at Context.<anonymous> (test/fixtures/error-diff.test.js:8:12)
      at processImmediate (internal/timers.js:439:21)

  2) error-diff
       compare different type:
     AssertionError: ==
      at Context.<anonymous> (test/fixtures/error-diff.test.js:12:12)
      at processImmediate (internal/timers.js:439:21)

  3) error-diff
       compare same content objects:

      AssertionError: ==
      + expected - actual


      at Context.<anonymous> (test/fixtures/error-diff.test.js:18:12)
      at processImmediate (internal/timers.js:439:21)

  4) error-diff
       compare different content objects deeply:

      AssertionError: deepEqual
      + expected - actual

       {
      -  "a": 1
      +  "a": 2
       }

      at Context.<anonymous> (test/fixtures/error-diff.test.js:24:12)
      at processImmediate (internal/timers.js:439:21)

  5) error-diff
       display add diff:

      AssertionError: deepEqual
      + expected - actual

       {
         "a": 1
      +  "b": 2
      +  "c": 3
       }

      at Context.<anonymous> (test/fixtures/error-diff.test.js:30:12)
      at processImmediate (internal/timers.js:439:21)

  6) error-diff
       display delete diff:

      AssertionError: deepEqual
      + expected - actual

       {
         "a": 1
      -  "b": 2
      -  "c": 3
       }

      at Context.<anonymous> (test/fixtures/error-diff.test.js:36:12)
      at processImmediate (internal/timers.js:439:21)

  7) error-diff
       different multiple blocks:

      AssertionError: deepEqual
      + expected - actual

       {
      -  "a": true
      +  "a": false
         "b": {
           "c": "A"
           "d": 1
           "f": {
--
         "l": 2
         "m": 3
         "n": [
           4
      -    5
      +    6
         ]
       }

      at Context.<anonymous> (test/fixtures/error-diff.test.js:72:12)
      at processImmediate (internal/timers.js:439:21)

  8) error-diff
       different multiple blocks but nearly:

      AssertionError: deepEqual
      + expected - actual

       {
      -  "a": true
      +  "a": false
         "b": {
      -    "c": "ABC"
      +    "c": "aaa"
           "d": 1
         }
       }

      at Context.<anonymous> (test/fixtures/error-diff.test.js:78:12)
      at processImmediate (internal/timers.js:439:21)



